internal/datastore/common/chunkbytes_test.go

```diff
@@ -0,0 +1,712 @@
+	require.NoError(t, err)
+
+	// Verify the DELETE query
+	require.Len(t, txn.capturedSQL, 1)
+	deleteSQL := txn.capturedSQL[0]
+	require.Contains(t, deleteSQL, "DELETE FROM test_table")
+	require.Contains(t, deleteSQL, "WHERE name = ?")
+	require.Equal(t, []any{"test-key"}, txn.capturedArgs[0])
+}
+
+func TestDeleteChunkedBytes_InsertWithTombstones(t *testing.T) {
+	txn := &fakeTransaction{}
+	executor := &mockExecutor{transaction: txn}
+	chunker := MustNewSQLByteChunker(SQLByteChunkerConfig[uint64]{
+		TableName:         "test_table",
+		NameColumn:        "name",
+		ChunkIndexColumn:  "chunk_index",
+		ChunkDataColumn:   "chunk_data",
+		MaxChunkSize:      10,
+		PlaceholderFormat: sq.Question,
+		Executor:          executor,
+		WriteMode:         WriteModeInsertWithTombstones,
+		CreatedAtColumn:   "created_at",
+		DeletedAtColumn:   "deleted_at",
+		AliveValue:        ^uint64(0), // max uint64
+	})
+
+	deletedAt := uint64(456)
+	err := chunker.DeleteChunkedBytes(context.Background(), "test-key", deletedAt)
+	require.NoError(t, err)
+
+	// Verify the UPDATE query (tombstone)
+	require.Len(t, txn.capturedSQL, 1)
+	updateSQL := txn.capturedSQL[0]
+	require.Contains(t, updateSQL, "UPDATE test_table")
+	require.Contains(t, updateSQL, "SET deleted_at = ?")
+	require.Contains(t, updateSQL, "WHERE name = ?")
+	require.Contains(t, updateSQL, "AND deleted_at = ?")
+	require.Equal(t, []any{deletedAt, "test-key", ^uint64(0)}, txn.capturedArgs[0])
+}
+
+func TestDeleteChunkedBytes_EmptyName(t *testing.T) {
+	txn := &fakeTransaction{}
+	executor := &mockExecutor{transaction: txn}
+	chunker := MustNewSQLByteChunker(SQLByteChunkerConfig[uint64]{
+		TableName:         "test_table",
+		NameColumn:        "name",
+		ChunkIndexColumn:  "chunk_index",
+		ChunkDataColumn:   "chunk_data",
+		MaxChunkSize:      10,
+		PlaceholderFormat: sq.Question,
+		Executor:          executor,
+		WriteMode:         WriteModeDeleteAndInsert,
+	})
+
+	err := chunker.DeleteChunkedBytes(context.Background(), "", 0)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "name cannot be empty")
+}
+
+func TestChunkData(t *testing.T) {
+	executor := &mockExecutor{}
+	chunker := MustNewSQLByteChunker(SQLByteChunkerConfig[uint64]{
+		TableName:         "test_table",
+		NameColumn:        "name",
+		ChunkIndexColumn:  "chunk_index",
+		ChunkDataColumn:   "chunk_data",
+		MaxChunkSize:      5,
+		PlaceholderFormat: sq.Question,
+		Executor:          executor,
+		WriteMode:         WriteModeDeleteAndInsert,
+	})
+
+	tests := []struct {
+		name          string
+		data          []byte
+		expectedCount int
+		expectedSizes []int
+	}{
+		{
+			name:          "empty data",
+			data:          []byte{},
+			expectedCount: 0,
+			expectedSizes: []int{},
+		},
+		{
+			name:          "exact chunk size",
+			data:          []byte("12345"),
+			expectedCount: 1,
+			expectedSizes: []int{5},
+		},
+		{
+			name:          "multiple exact chunks",
+			data:          []byte("1234567890"),
```

Comment:
Do these happen to line up because utf-8 is one byte per char for ascii characters?